### DISCUSSION TITLE
How to reset Supabase database?

### DISCUSSION THREAD

**Discussion by nicholaschiang** [supabase/supabase#1829] [Questions]
How do I perform a full-reset of any changes that I've made to the Supabase-managed PostgreSQL database?

I found [this answer on StackOverflow](https://stackoverflow.com/a/13823560/10023158) and it worked fine:

```pgsql
drop schema if exists public cascade;
create schema public;
grant all on schema public to postgres;
grant all on schema public to public;
```

However, I think that ☝️ somehow broke a Supabase configuration that enables the API because I now get:

```
error: permission denied for table users
```

whenever I try to insert into a table (that I successfully re-created via the SQL editor after resetting my database).

Is there a way that I can restore a project's database to the state it was in right after Supabase finished setting it up? It'd be really nice to have a SQL script (like the "Create a table" or "Add column" scripts) already included in the "Welcome" tab for this. Or, a button in the project settings' danger zone that lets me reset my database.

**Comment by soedirgo** [comment]
Hey @nicholaschiang! If you have a backup in Database > Settings > Backups you might be able to restore your database, otherwise you'll have to resort to creating a new project.

**Reply by nicholaschiang** [reply]
@soedirgo That's not the problem. I want to delete everything but I don't want to mess up anything that Supabase sets up.

**Reply by soedirgo** [reply]
Hmm, wouldn't restoring a backup achieve exactly that? (Assuming you have a backup of the `public` schema on project creation.)

**Comment by steve-chavez** [comment]
> grant all on schema public to postgres;
> grant all on schema public to public;

Both of these grants only work for all tables that are currently created. If you create new ones, they will be restricted for those roles.

If you want roles to have privileges on all future tables, you can run:

```sql
alter default privileges in schema public grant all on tables to postgres, anon, authenticated, service_role;
```

(The above is specified on the [supabase cli schema](https://github.com/supabase/cli/blob/main/src/templates/init/docker/postgres/00-initial-schema.sql#L22-L25))

**Comment by TimDev9492** [comment]
I'm very frustrated about there not being an option to reset the database of a project. Bumping this thread.

**Reply by juanvilladev** [reply]
If you're using the CLI for developing, you can simply run 
```
supabase reset
```
The db will restart without any tenanted data. Not sure what use case there is for resetting your remote db?

**Reply by kaevupoiss** [reply]
@juanvilladev I want to keep iterating on my staging database with rewriting old migrations. I need the equivalent of `supabase db reset` for the remote db. I only care about the seed data and schema while in pre-prod development.

**Reply by tomelliot** [reply]
FYI according to https://github.com/orgs/supabase/discussions/13832#discussioncomment-5997090 you can use `supabase db reset --db-url <DB_URL>`.

I just tried it and it with a basic test and seems to work, although it might be only for the public schema - there's still data maintained in the auth.users table.

**Reply by tomelliot** [reply]
Ah I read a bit further. Here's more info for you:

`supabase db reset --linked` is [the supported command](https://github.com/supabase/cli/issues/480#issuecomment-1565196571) to reset the remote db for a project. 

From [the PR](https://github.com/supabase/cli/pull/1105):
>     ignores all supabase managed schemas, including auth, storage, extensions, etc.
>     only drop objects in public schema to preserve default schema privileges for supabase_admin
>     applies all location migrations and seeds remote database
